Library/Util.ts

```diff
@@ -151,5 +154,27 @@ class Util {
 
         return map;
     }
+
+    /**
+     * Checks if a request url is not on a excluded domain list 
+     * and if it is safe to add correlation headers (x-ms-request-source-ikey, x-ms-request-target-ikey)
+     */
+    public static canIncludeCorrelationHeader(client: Client, url: string) {
+        let excludedDomains = client && client.config && client.config.correlationHeaderExcludedDomains;
+        if (!excludedDomains || excludedDomains.length == 0 || !url) {
+            return true;
+        }
+
+        for (let i = 0; i < excludedDomains.length; i++) {
+            let regex = new RegExp(excludedDomains[i].replace(/\./g,"\.").replace(/\*/g,".*"));
+            let matches = url.match(regex);
```

Comment:
It would be a little more efficient to use the `RegExp.test()` method here, since that just returns a boolean result instead of creating an array of match objects.